pkg/engine/mutate/strategicPreprocessing_test.go

```diff
@@ -27,6 +27,56 @@ func Test_preProcessStrategicMergePatch(t *testing.T) {
 	}
 }
 
+func Test_preProcessStrategicMergePatch_Equality(t *testing.T) {
+	rawPolicy := []byte(`{
+		"spec": {
+			"volumes": {
+				"=(emptyDir)": {
+					"+(medium)": "Memory"
+				}
+			}
+		}
+	}`)
+
+	rawResource := []byte(`{
+		"spec": {
+			"containers": [
+				"image": "foo"
+			],
+			"volumes": [
+				{
+					"name": "cache-volume",
+					"emptyDir": {}
+				},
+				{
+					"name": "cache-volume2",
+					"emptyDir": {}
+				},
+				{
+					"name": "baar",
+					"emptyDir": {
+						"medium": "Memory"
+					}
+				}
+			]
+		}
+	}`)
+
+	expected := `{"spec": {"volumes": ["{"emptyDir": {"medium": "Memory"}}", "{"emptyDir": {"medium": "Memory"}}", "{"emptyDir": {}}"]}}`
```

Comment:
@kacejot - have you verified it in a live cluster? What's the patched object if we do not have tag `name` for each volume?

Reply:
@kacejot - the reason we want to preserve the tag `name` is that it is used as a merge key to do a strategic merge patch. If it's missing, they will be treated as new elements and append to the list.

Reply:
@realshuting OK. So this issue is not fixed then. I will apply edits with the `name` tag and notify you